### PR TITLE
Fixing elementwise absolute value operation on UserTypes

### PR DESCRIPTION
**Pull request by cmikida2**
An effort to fix a previous erroneous attempt at elementwise absolute values (see https://github.com/inducer/dagrt/pull/20). This new PR features tests that actually check that the absolute values are being properly calculated using this builtin for scalars, arrays, and UserTypes.

### CODE REVIEW

File: test/test_codegen_fortran.py [hunk 1]
```diff
@@ -150,19 +150,35 @@ def test_self_dep_in_loop():
...
+    pass
+
+
 def test_elementwise_abs():
     with CodeBuilder(name="primary") as cb:
         cb("y", "<func>f(0, <state>ytype)")
         cb("<state>ytype", "y")
         # Test new builtin on a usertype.
         cb("z", "<builtin>elementwise_abs(<state>ytype)")
+        with cb.if_("z[1] > 2"):
```
Comment (inducer) [inline test/test_codegen_fortran.py:163]: It looks like this is indexing into a user type, which wouldn't be allowed. Can you help me understand? If you'd like to assert something about the internal state of a user type, it should be done in Fortran.

File: test/test_element_abs.f90 [hunk 2]
```diff
@@ -20,7 +20,7 @@ program test_element_abs
 
 
   do i = 1, 100
-    y0 = i
+    y0(i) = i
```
Comment (inducer) [inline test/test_element_abs.f90:23]: Wouldn't it be better to include negative numbers in a test of `abs`?

**Review by inducer — COMMENTED** [review]
I said I'd left comments during the meeting. Turns out that was only half-true. Typed them in, but not yet submitted them. Sorry!

File: test/test_element_abs.f90 (hunk 2 above)
Comment (cmikida2) [reply test/test_element_abs.f90:23]: Yep - this usertype goes into a function first that multiplies it by -2. Then it's checked, so we are checking negative numbers.

File: test/test_codegen_fortran.py (hunk 1 above)
Comment (cmikida2) [reply test/test_codegen_fortran.py:163]: Yeah, this is a fair point. Obviously given the inadequacy of the previous version of the test I wanted to make sure I'm checking the result when the function is applied to scalars, arrays, and usertypes, but without indexing into the UserType (which is just an array) I'm not sure how to check it. Can you explain what you mean by the last part of your comment?

Comment (cmikida2) [reply test/test_codegen_fortran.py]: Nvm - I *think* I did what makes more sense in 0237e4d. (Let me know if this isn't what you meant)

File: test/test_element_abs.f90 (hunk 2 above)
Comment (inducer) [reply test/test_element_abs.f90:25]: What I meant was: maybe include both positive and negative numbers? :)

Comment (inducer) [reply test/test_element_abs.f90:25]: Ah, you know what, this is good enough.

**Comment by inducer** [comment]
LGTM, thanks!

**Pull request merged**